sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/ied/InputsAdapter.java

```diff
@@ -381,18 +377,32 @@ private static String generateDataSetSuffix(TExtRef extRef, DataAttributeRef sou
                 + (isBayInternal ? "I" : "E");
     }
 
-    private Optional<SclReportItem> removeFilteredSourceDas(TExtRef extRef, final Set<DataAttributeRef> sourceDas) {
+    private Optional<SclReportItem> removeFilteredSourceDas(TExtRef extRef, final Set<DataAttributeRef> sourceDas, Set<FcdaCsvHelper.FcdaForDataSets> allowedFcdas) {
         sourceDas.removeIf(da -> da.getFc() != TFCEnum.MX && da.getFc() != TFCEnum.ST);
         return switch (extRef.getServiceType()) {
             case GOOSE, SMV -> {
-                sourceDas.removeIf(Predicate.not(FcdaCandidates.SINGLETON::contains));
+                sourceDas.removeIf(Predicate.not(dataAttributeRef -> isFcdaAllowed(dataAttributeRef, allowedFcdas)));
                 yield Optional.empty();
             }
             case REPORT -> removeFilterSourceDaForReport(extRef, sourceDas);
             default -> fatalReportItem(extRef, String.format(MESSAGE_INVALID_SERVICE_TYPE, extRef.getServiceType()));
         };
     }
 
+    private boolean isFcdaAllowed(DataAttributeRef dataAttributeRef, Set<FcdaCsvHelper.FcdaForDataSets> allowedFcdas) {
+        String lnClass = dataAttributeRef.getLnClass();
+        String doName = dataAttributeRef.getDoName().toStringWithoutInst();
+        String daName = dataAttributeRef.getDaName().toString();
+        String fc = dataAttributeRef.getFc().value();
+        if (StringUtils.isBlank(lnClass) || StringUtils.isBlank(doName) || StringUtils.isBlank(daName) || StringUtils.isBlank(fc)) {
```

Comment:
not checked in xsd ?

Reply:
No, as source (Set<DataAttributeRef> sourceDas) is filled with ExtRefs datas and when extref is not binded  this parameters can be null